sdk/keyvault/azure-security-keyvault-keys/test/samples/sample2-backup-and-restore/sample2_backup_and_restore.cpp

```diff
@@ -70,8 +70,22 @@ int main()
     // You only need to wait for completion if you want to purge or recover the key.
     operation.PollUntilDone(std::chrono::milliseconds(2000));
     keyClient.PurgeDeletedKey(rsaKeyName);
-    // let's wait for one minute so we know the key was purged.
-    std::this_thread::sleep_for(std::chrono::seconds(60));
+
+    // Let's wait for a bit so we know the key was purged.
+    try
+    {
+      // To check if the key was purged we attempt to get the key from the Key Vault.
+      // If we get an exception, the key was purged.
+      // If not, we wait a bit more,since the key is in the purge process.
+      while (!keyClient.GetDeletedKey(rsaKeyName).Value.Name().empty())
+      {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+      }
+    }
```

Comment:
I like the quicker loop, should in most cases make the tests faster.  Lets put a hard limit of 5 minutes and fail if it goes that long.  That way we don't spin forever.

Reply:
Would it make sense to have a Context variable with a timeout and then specify the context to the GetDeletedKey call?

Reply:
true that